tests/integration/test_charm.py

```diff
@@ -96,26 +80,54 @@ async def test_database_is_up(ops_test: OpsTest, series: str):
     action = await action.wait()
     password = action.results["postgres-password"]
 
-    # Testing the connection to each PostgreSQL instance.
-    for unit in ops_test.model.applications[application_name].units:
-        # List clusters to assert there is only one cluster per unit.
-        result = await run_command_on_unit(unit, "pg_lsclusters --no-header")
-        clusters = result.splitlines()
-        assert len(clusters) == 1
-
-        cluster_data = clusters[0].split()
-        # Check for correct cluster version.
-        assert cluster_data[0] == POSTGRESQL_VERSIONS[series]
-        # And check cluster status (online or down).
-        assert cluster_data[3] == "online"
-
-        # Then, test the connection.
-        host = unit.public_address
-        logger.info("connecting to the database host: %s", host)
-        connection = psycopg2.connect(
-            f"dbname='postgres' user='postgres' host='{host}' password='{password}' connect_timeout=1"
-        )
+    # Connect to PostgreSQL.
+    host = ops_test.model.units.get(f"{application_name}/0").public_address
+    logger.info("connecting to the database host: %s", host)
+    with psycopg2.connect(
+        f"dbname='postgres' user='postgres' host='{host}' password='{password}' connect_timeout=1"
+    ) as connection:
         assert connection.status == psycopg2.extensions.STATUS_READY
-        # TODO: check for correct number of members (master and standby replicas)
-        # and replication status when replication is implemented.
-        connection.close()
+
+        # Retrieve settings from PostgreSQL pg_settings table.
+        # Here the SQL query gets a key-value pair composed by the name of the setting
+        # and its value, filtering the retrieved data to return only the settings
+        # that were set by Patroni.
+        cursor = connection.cursor()
+        cursor.execute(
+            """SELECT name,setting
+                FROM pg_settings
+                WHERE name IN
+                ('data_directory', 'cluster_name', 'data_checksums', 'listen_addresses');"""
+        )
```

Comment:
You've forgotten to close `cursor`

Reply:
Thanks for catching this, Will. Fixed on https://github.com/canonical/postgresql-operator/pull/8/commits/d2d0915c789894145f29c296de6b4ec5492cf382.